src/main/java/seedu/address/model/VersionedAddressBook.java

```diff
@@ -0,0 +1,144 @@
+package seedu.address.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.address.model.supplier.Supplier;
+import seedu.address.model.supplier.UniqueSupplierList;
+
+/**
+ * An {@code AddressBook} that keeps track of its history. Snapshots of its state are done based on external commands.
+ */
+public class VersionedAddressBook extends AddressBook implements Version<AddressBook> {
+    private AddressBook currentState;
+    private int statePointer;
+    private List<AddressBook> stateList;
+
+    public VersionedAddressBook() {
+        statePointer = -1;
+        stateList = new ArrayList<>();
+        currentState = new AddressBook();
+        commit();
+    }
+
+    /**
+     * Creates a VersionedAddressBook using the {@code Supplier}s in the {@code toBeCopied}.
+     */
+    public VersionedAddressBook(ReadOnlyList<Supplier> toBeCopied) {
+        statePointer = -1;
+        stateList = new ArrayList<>();
+        currentState = new AddressBook(toBeCopied);
+        commit();
+    }
+
+    //// list overwrite operations
```

Comment:
For visual demarcation of code should we follow the style from Model.java, where they use "//=========== UserPrefs ======" instead?

Reply:
I think your suggestion sounds good, I'll apply it to all these cases